pkg/vetter/util/mtlspolicy/authPolicy.go

```diff
@@ -122,85 +160,254 @@ func (ap *AuthPolicies) ByPort(s Service, port uint32) []*authv1alpha1.Policy {
 	return p
 }
 
-// This returns all the policies for a cluster that whose target is a port, regardless of service or namespace
-func (ap *AuthPolicies) GetClusterPortPolicies() []*authv1alpha1.Policy {
-	portPolicies := []*authv1alpha1.Policy{}
-	for _, polsByNamePortMap := range ap.port {
-		for _, polsByPortMap := range polsByNamePortMap {
-			for _, policies := range polsByPortMap {
-				portPolicies = append(portPolicies, policies...)
+// ForEachPolByPort takes a callback and applies it to a range of policies by port
+func (ap *AuthPolicies) ForEachPolByPort(s Service, cb func(s Service, port uint32, policies []*authv1alpha1.Policy)) error {
+
+	ns, ok := ap.port[s.Namespace]
+	if !ok {
+		return errors.New("No policies for namespace that specify a port")
+	}
+	n, ok := ns[s.Name]
+	if !ok {
+		return errors.New("No policies for service that specify a port")
+	}
+	for port, policies := range n {
+		cb(s, port, policies)
+		return nil
+	}
+	return nil
+}
+
+// getMTLSBool returns a bool and error from the 4 possible enum mTls states.
+// Mixed counts as enabled since it allows enabled traffic, but it returns an error in case the caller needs to know if the true status means it's enabled-only, or enabled in a way that allows other traffic.
+// Unknown counts as disabled since we cannot tell the caller that the status is mTls enabled. It returns an error in case the caller needs to know if the false status means that the false status is actually bogus because we we unable to determine the mTls status.
+func getMTLSBool(mtlsState MTLSSetting, policy *authv1alpha1.Policy) (bool, *authv1alpha1.Policy, error) {
+	// pass in the policy to maintain the structure of returns for callers pre-refactor.
+	switch checkState := mtlsState; {
+	case checkState == MTLSSetting_ENABLED:
+		return true, policy, nil
+	case checkState == MTLSSetting_UNKNOWN:
+		return false, policy, errors.New("mTLS status is Unknown")
```

Comment:
Before TLSByPort might have returned an error "Conflicting Policies for Port" and now I think that code path ends up here, which squashes the "Conflicting Policies for Port" error and replaces it with "mTLS status is Unknown".  Is that intentional?  Will the vetter still work and return a useful warning to the user?